### ISSUE TITLE
Add RabbitMQ as an event source

### ISSUE THREAD

**Issue by sc68cal** [ansible/event-driven-ansible#182]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/ansible-rulebook/issues) for duplicates.
- [X] I understand that ansible-rulebook is open source software provided for free and that I might not receive a timely response.

### Feature type

New Feature

### Feature Summary

It would be great to have RabbitMQ as an event source

**Comment by Alex-Izquierdo** [comment]
Hello @sc68cal 

After careful consideration, we have concluded that a plugin for RabbitMQ does not align with the scope of our official collection. For your reference, you may find [this pull request](https://github.com/ansible/event-driven-ansible/pull/59) insightful.

**Comment by sc68cal** [comment]
@Alex-Izquierdo can you share why this does not align with the scope of this official collection? Two separate contributors have created PRs that implemented this feature, so it is in demand.

**Comment by Alex-Izquierdo** [comment]
Hi @sc68cal 
Eda-collection is intended to provide the initial resources for getting started with the eda-server, as well as the integrations required for its features, by delegating third-party integrations to the collections developed by the ansible community.

**Comment by Alex-Izquierdo** [comment]
@sc68cal I'm really sorry for having seemed abrupt. That was never my intention and I should not have closed this issue. Indeed we are open to every discussion or suggestion from the community. I hope you will disregard this unfortunate mistake. 

I understand your point of view and I think it's quite reasonable. Keep in mind that with each plugin added to this collection, there is a commitment to maintenance and therefore an associated effort. As I have previously explained, the purpose of this collection is to provide the basic set of tools for starting with eda-server and those integrations necessary for its functionality. It is not our intention to turn this collection into an all-in-one that contains all potential plugins, nor do we have the resources to maintain such a set.

Indeed, I think you are right that creating a single collection for a plugin can be overkill. Instead, I can suggest that you propose this extension to the [rabbitmq community collection](https://github.com/ansible-collections/community.rabbitmq) that already exists, which would also be quite coherent in terms of making accessible a collection that provides all the necessary tools for integration with the ansible ecosystem and rabbitmq (modules, eda extensions, etc.). That is why eda plugins are implemented as an extension inside collections. If users intend to use rabbitmq as a source of events for eda, it makes sense that they would also want to perform other configurations on queues for example, so it would be logical to do everything from the same collection.

Another option that we considered some time ago was to promote the creation of a plugin collection for eda plugins exclusively maintained by the community, something equivalent to [community.general](https://github.com/ansible-collections/community.general). We eventually discarded it because at that time there didn't seem to be enough traction or contributions to justify its existence. But I am sure that it is an option that can be reconsidered if the situation changes. Still, we continue to have the doubt of what is more practical and intuitive for users. Whether a repository that aggregates plugins of various natures is more consistent, or if it is more consistent for each collection to have the eda plugins related to the scope of that collection (a rabbitmq collection, a kubernetes collection, etc.)

**Comment by sc68cal** [comment]
> indeed, I think you are right that creating a single collection for a plugin can be overkill. Instead, I can suggest that you propose this extension to the [rabbitmq community collection](https://github.com/ansible-collections/community.rabbitmq) that already exists, which would also be quite coherent in terms of making accessible a collection that provides all the necessary tools for integration with the ansible ecosystem and rabbitmq (modules, eda extensions, etc.).

In theory this would be acceptable, however the rabbitmq community collection uses `pika` as its RabbitMQ library, which does not work with asyncio. I would be adding a brand new dependency to `aiorabbit` which could turn into a large bikeshedding debate.

In addition, it would be easy for the rabbitmq community collection maintainers to turn around and use the same argument as yours, where dealing with event driven ansible is out of scope for their collection, and why isn't it in the EDA collection?

Basically, I think that for now, EDA is the location where this should go, since it would co-exist quite well with the other event sources that are being maintained.

In addition, I see that a PostgreSQL event source has been proposed in another PR, but I do not see the same pushback there, where the event source plugin should be sent over to the community.postgresql collection. 

Can you help shed light on this situation for me?

**Comment by fkuep** [comment]
@Alex-Izquierdo Which [standards-conforming|http://docs.oasis-open.org/amqp/] event distribution machanism could be considered basic enough to be considered for inclusion into and to be supported by eda.
[mqtt|https://datatracker.ietf.org/doc/html/rfc9431]? 
There seems to be a [way|https://github.com/eclipse/paho.mqtt.python/blob/master/tests/testsupport/broker.py] to create an integration test.  
_We need a sign, that eda has a valid scope outside product release cycles. Sitting outside with webhooks does not give me the feeling of participation either._

**Comment by Alex-Izquierdo** [comment]
Hello @fkuep 
I think it's not just a matter of determining which mechanism is basic enough to be included (I agree that MQTT is definitely one of the most relevant ones). The main point concerns the purpose and scope of this collection, which I explained in previous comments.

**Comment by fkuep** [comment]
Hello, @Alex-Izquierdo I understand and comming from the  [rulebook doc](https://ansible.readthedocs.io/projects/rulebook/en/stable/sources.html#best-practices-and-patterns) I was of the impression :

- This is the repository one starts to learn with.
- one should best start with "event-bus-like event distribution systems" for activating rulebooks.
- the new and shiny event driven ansible thingy is going to be big also for the ansible community.

What I want from the Team working on this repository is either to develop an outline, how this can be used as independant reference implementation or ..
Have the ansible-rulebook documentation changed to reflect that:
_This eda repo is a component of IBM Cloud offerings, we have this sandbox and we don´t share the shovel. Thanks for Your bug-fixes!_

If it is the first one, **one** event mechanism that You consider best-practise should be
- part of this _or an other reference implementation_ repository, 
- fairly easy to be installed (e.g. not three nodes, zookeeper and java) 
- **Standards conformant**
- tested with ci.
- kept around , at least until other low-effort/high quality reference event sources are available.

So please Alex ask around for us:  We would like to know, if we are invited to use and promote this or if we are better off using https://github.com/adnanh/webhook to avoid technical dept.

**Comment by tarmael** [comment]
I'm also getting started with Ansible EDA, my first use-case for EDA is to retrieve messages from a queue and perform automations from there.

I'm surprised to learn Azure Service Bus is in the official library set, but RabbitMQ is not.

If the consideration for this plugin is down to a clash on the python libraries then perhaps EDA plugins should be maintained in their individual product namespaces and this repository deprecated.

**Comment by Alex-Izquierdo** [comment]
@tarmael Azure Service Bus was added in the early stages of the development of the collection, prior to decide what would be the scope of the collection. We keep it for now for backward compatibility but it will be deprecated eventually and moving it out to the eda community collection that we are preparing and we expect to have it ready soon.
